test/clj/puppetlabs/trapperkeeper/services/webserver/jetty9_core_test.clj

```diff
@@ -174,3 +177,126 @@
+        ;; and I didn't see any other way to pull the queue back from
+        ;; the thread pool.
+        get-queue-method (-> thread-pool
+                             (.getClass)
+                             (.getDeclaredMethod "getQueue" nil))
+        _                (.setAccessible get-queue-method true)
+        queue            (.invoke get-queue-method thread-pool nil)]
+    {:thread-pool thread-pool
+     :queue queue}))
+
+(deftest create-server-tests
+  (testing "thread pool given proper values"
+    (testing "max threads passed through"
+      (dotimes [x 5]
+        (let [{:keys [thread-pool]}
+          (get-thread-pool-and-queue-for-create-server x 1)]
+          (is (= x (.getMaxThreads thread-pool))
+              "Unexpected max threads for queue"))))
+
+    (testing "default idle timeout passed through"
+      (let [{:keys [thread-pool]}
+             (get-thread-pool-and-queue-for-create-server 42 1)]
+        (is (= jetty/default-queue-idle-timeout
+               (.getIdleTimeout thread-pool)))))
+
+    (testing "when queue-max-size less than default-queue-min-threads"
+      (let [max-threads    23
+            queue-min-size (dec jetty/default-queue-min-threads)
+            {:keys [thread-pool queue]}
+              (get-thread-pool-and-queue-for-create-server max-threads
+                                                           queue-min-size)]
+        (is (= max-threads (.getMaxThreads thread-pool))
+            "Unexpected max threads for thread pool")
+        (is (= queue-min-size (.getMinThreads thread-pool))
+            "Unexpected min threads for queue")
+        (is (= queue-min-size (.getCapacity queue))
+            "Unexpected initial capacity for queue")
+        (is (= queue-min-size (.getMaxCapacity queue))
+            "Unexpected max capacity for queue")))
+
+    (testing "when queue-max-size greater than default-queue-min-threads"
+      (let [max-threads    42
+            queue-min-size (inc jetty/default-queue-min-threads)
+            {:keys [thread-pool queue]}
+            (get-thread-pool-and-queue-for-create-server max-threads
+                                                         queue-min-size)]
+        (is (= max-threads (.getMaxThreads thread-pool))
+            "Unexpected max threads for thread pool")
+        (is (= jetty/default-queue-min-threads (.getMinThreads thread-pool))
+            "Unexpected min threads for queue")
+        (is (= jetty/default-queue-min-threads (.getCapacity queue))
+            "Unexpected initial capacity for queue")
+        (is (= queue-min-size (.getMaxCapacity queue))
+            "Unexpected max capacity for queue")))))
+
+(defn get-responses
+  [num-requests port]
+  (doall (for [_ (range num-requests)]
+           (future
+             (http-sync/get
+               (str "http://localhost:"
+                    port
+                    "/hello")
+               {:as :text})))))
+
+(deftest queue-max-size-tests
```

Comment:
I struggled mightily with trying to create a more "end-to-end" level test that proves out this functionality.  These tests definitely have race conditions in them - e.g., with the "sleep" statements and guesses at a sufficient number of concurrent requests that would need to be made in order to tickle the upper limit on the queue.  I haven't been able to come up with a safer set of tests that exercises the functionality in a meaningful way, though.  I ultimately think it might be better to just remove these tests altogether but wanted to get thoughts from others first.